### PR TITLE
Standalone Lite: DataParallel Strategy

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Adds the DP strategy for Lite. It is very similar to the one in PL. It will be one of the easiest to "migrate" when we later make PL depend on the implementation in Lite.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: src/lightning_lite/strategies/dp.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    @property
+    def world_size(self) -> int:
+        return 1
+
+    def setup_module(self, module: Module) -> DataParallel:
+        """Wraps the given model into a :class:`~torch.nn.parallel.DataParallel` module."""
+        return DataParallel(module=module, device_ids=self.parallel_devices)
+
+    def module_to_device(self, module: Module) -> None:
+        module.to(self.root_device)
```
Comment (carmocca) [inline src/lightning_lite/strategies/dp.py:74]: Noticing now that these are `module` instead of `model` and that the APIs are slightly different here vs in PL. Will they be reconcilliated eventually?

File: src/lightning_lite/strategies/dp.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    def local_rank(self) -> int:
+        return 0
+
+    @property
+    def node_rank(self) -> int:
+        return 0
+
+    @property
+    def world_size(self) -> int:
+        return 1
```
Comment (carmocca) [inline src/lightning_lite/strategies/dp.py:67]: Technically these are not necessary since `cluster_environment=None` so the parent's implementations are enough

```suggestion
```

File: src/lightning_lite/strategies/dp.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    def setup_module(self, module: Module) -> DataParallel:
+        """Wraps the given model into a :class:`~torch.nn.parallel.DataParallel` module."""
+        return DataParallel(module=module, device_ids=self.parallel_devices)
+
+    def module_to_device(self, module: Module) -> None:
+        module.to(self.root_device)
+
+    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
```
Comment (carmocca) [inline src/lightning_lite/strategies/dp.py:76]: Lite should remove the dataloader_idx? The base class does not have it

```suggestion
    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None) -> Any:
```

**Review by carmocca — APPROVED** [review]

File: src/lightning_lite/strategies/dp.py (hunk 3 above)
Comment (awaelchli) [reply src/lightning_lite/strategies/dp.py:61]: Thank you! Small mishaps like these happen, until we have the full integration with all tests running. Fingers crossed!

File: src/lightning_lite/strategies/dp.py (hunk 1 above)
Comment (awaelchli) [reply src/lightning_lite/strategies/dp.py:58]: Yes, purposefully named module for clarity. 
PL will eventually also call this method, and feed into it the LightningModule (or wrapped LM if applicable). 

I felt comfortable doing the name change, especially because we needed to do another change anyway: The module as input argument. This is missing in PL's version    ` def model_to_device(self) -> None:`

**Review by Borda — APPROVED** [review]
